docs/design/secrets.md

```diff
@@ -0,0 +1,187 @@
+    pods, so that I can keep the configuration for my applications separate from the images that
+    use them:
+    1.  As a cluster operator, I want to allow a pod to access the Kubernetes master using a custom
+        .kubeconfig, so that I can securely reach the master
+    2.  As a cluster operator, I want to allow a pod to access the DockerHub using credentials from a
+        .dockercfg file, so that pods can push images to the DockerHub
+    3.  As a cluster operator, I want to allow a pod to access a GitHub repository using SSH keys, so
+        that I can push and fetch to and from the repository
+2.  As a user, I want to allow pods to consume supplemental information about services such as
+    username and password which should be kept secret, so that I can share secrets about a service
+    amongst the pods in my application securely
+
+### Use-Case: Configuration artifacts
+
+Many configuration files contain secrets intermixed with another configuration information.  For
+example, a user's application may contain a properties file than contains database credentials,
+SaaS API tokens, etc.  Users should be able to consume configuration artifacts in their pods and
+be able to control the path on the containers filesystem where the artifact will be presented.
+
+### Use-Case: Metadata about services
+
+Most pieces of information about how to use a service are secrets.  For example, a service that
+provides a mysql database needs to provide the username, password, and database name to consumers
+so that they can authenticate and use the correct database. Pods consuming the mysql service would
+also consume the secrets associated with the mysql service.
+
+## Consuming secrets as environment variables
+
+Some images will expect to receive configuration items as environment variables instead of files.
+We should consider what the best way to allow this is; there are a few different options:
+
+1.  Force the user to adapt files into environment variables.  Users can store secrets that need to
+    be presented as environment variables in a format that is easy to consume from a shell:
+
+        $ cat /etc/secrets/my-secret.txt
+        export MY_SECRET_ENV=MY_SECRET_VALUE
+
+    The user could `source` the file at `/etc/secrets/my-secret` prior to executing the command for
+    the image either inline in the command or in an init script, 
+
+2.  Give secrets an attribute that allows users to express the intent that the platform should
+    generate the above syntax in the file used to present a secret.  The user could consume these
+    files in the same manner as the above option.
+
+3.  Give secrets attributes that allow the user to express that the secret should be presented to
+    the container as an environment variable.  The container's environment would contain the
+    desired values and the software in the container could use them without accomodation the
+    command or setup script.
```

Comment:
This requires the least change in the container, but also exposes the secret value in the docker container definition (`docker inspect foo` would show the envvar value)